crates/bitwarden-ipc/src/ipc_client.rs

```diff
@@ -38,66 +39,81 @@ where
             .await
     }
 
+    /// Receive a message
     pub async fn receive(
         &self,
     ) -> Result<IncomingMessage, ReceiveError<Crypto::ReceiveError, Com::ReceiveError>> {
         self.crypto
             .receive(&self.communication, &self.sessions)
             .await
     }
+
+    /// Receive a message, skipping any messages that cannot be deserialized into the expected
+    /// payload type.
+    pub async fn receive_typed<Payload>(
+        &self,
+    ) -> Result<TypedIncomingMessage<Payload>, ReceiveError<Crypto::ReceiveError, Com::ReceiveError>>
+    where
+        Payload: TryFrom<Vec<u8>>,
+    {
+        loop {
+            let received = self
+                .crypto
+                .receive(&self.communication, &self.sessions)
+                .await?;
+            if let Ok(typed) = received.try_into() {
+                return Ok(typed);
+            }
```

Comment:
I wonder if we can come up with a way of validating the type other than just trying to parse it. As it is, we can't know for sure if we haven't received any message, or the message failed to parse correctly. Plus if we're passing keys as Vec<u8>, we might find ourselves accepting the wrong messages if they just happen to deserialize okay.

Only way to improve that I can think of would be to including the type as a string in the message, which probably would require also adding a `send_typed` equivalent.

Reply:
Hmm, very good point.

What about something like this? https://github.com/bitwarden/sdk-internal/pull/187/commits/1dd01642843290979fe9b6af89c3097ef702e7f2 We need to add Topic support for request/response anyways so

Reply:
Looks pretty good, though I would add some more documentation about what the `None` case does:
```
    /// Receive a message, filtering by topic.
    pub async fn receive_topic(
        &self,
        topic: Option<String>,
```

Reading that, I would assume that if I passed `None` to topic, no filtering would be done (so i would get all messages). What really happens is that I only get the messages without a topic.